src/platforms/android/usage/advanced-usage.mdx

```diff
@@ -0,0 +1,303 @@
+internally, and will therefore not work out of the box on newer Android devices which have their own conflicting pointer tagging.
+To prevent this, disable the [`allowNativeHeapPointerTagging`](https://developer.android.com/guide/topics/manifest/application-element#allowNativeHeapPointerTagging) configuration option.
+
+**Example AndroidManifest.xml**:
+
+```xml {filename:AndroidManifest.xml}
+<application
+    android:allowNativeHeapPointerTagging="false"
+    android:extractNativeLibs="true"
+    >
+</application>
+```
+
+To use the Android NDK in your native code, include the Sentry NDK libraries into your project so that the compiler can link the libraries during the build.
+
+To do so, use the AndroidNativeBundle Gradle plugin that copies the native libraries from the Sentry NDK into the location that can provide links via the CmakeLists.txt configuration file.
+
+First, we need to declare the dependency in the project build.gradle file:
+
+```groovy
+buildscript {
+    repositories {
+        jcenter()
+    }
+
+    dependencies {
+        // Add the line below, the plugin that copies the binaries
+        // https://github.com/howardpang/androidNativeBundle/releases
+        classpath 'com.ydq.android.gradle.build.tool:nativeBundle:{version}'
+    }
+}
+```
+
+Once the dependency is declared, we can use it in the application build.gradle:
+
+```groovy
+apply plugin: 'com.ydq.android.gradle.native-aar.import'
+```
+
+The last step is to update the CmakeLists.txt configuration to link the Sentry NDK libraries:
+
+```
+# include paths generated by androidNativeBundle
+include (${ANDROID_GRADLE_NATIVE_BUNDLE_PLUGIN_MK})
+# change native-lib to your native lib's name
+target_link_libraries(native-lib ${ANDROID_GRADLE_NATIVE_MODULES})
+```
+
+Now you can use the Sentry NDK API just by including the sentry.h in your code:
+
+```c
+#include <jni.h>
+#include <android/log.h>
+#include <sentry.h>
+
+#define TAG "sentry-android-demo"
+extern "C" JNIEXPORT jstring JNICALL
+
+Java_io_sentry_demo_NativeDemo_crash(JNIEnv *env, jclass cls) {
+    __android_log_print(ANDROID_LOG_WARN, "", "Capture a message.");
+    sentry_value_t event = sentry_value_new_message_event(
+            /*   level */ SENTRY_LEVEL_INFO,
+            /*  logger */ "custom",
+            /* message */ "Sample message!"
+    );
+    sentry_capture_event(event);
+}
+```
+
+To symbolicate the stack trace from native code, we need to have access to the debug symbols of your application. Please check the full documentation on [uploading files](/workflow/debug-files/#uploading-files) to learn more about the upload of the debug symbols.
+
+Example of uploading all your .so files:
+
+```bash
+sentry-cli login
+sentry-cli upload-dif -o {YOUR ORGANISATION} -p {PROJECT} build/intermediates/merged_native_libs/{buildVariant}
+```
+
+The Sentry Gradle Plugin can be [configured](#gradle-configuration) to upload the debug symbols and sources automatically.
+
+### Crash Reporting
```

Comment:
i didnt grok this section - why it exists or what its purpose is.

Reply:
this (Crash Reporting) should not live under advanced-usage for sure, Filip just wanted to explain how the capture exception flow works I guess, and also to point out the differences between the Java and sentry-native (NDK) layer, it's like an intro on how the SDK works